internal/internal.go

```diff
@@ -392,7 +411,7 @@ func (p *envoyExtAuthzGrpcServer) check(ctx context.Context, req interface{}) (*
 	}
 
 	if ctx.Err() != nil {
-		err = errors.Wrap(ctx.Err(), "check request timed out before query execution")
+		err = errors.Wrap(ctx.Err(), CheckRequestTimeoutErrorMessage)
```

Comment:
[nit] We should introduce our own error type here, and match on that type instead of the error message, please.

Reply:
@srenatus Thanks for the comment! I have tried to address this with minimal changes, please do review.

Reply:
Thanks! I'll review it tomorrow unless @ashutosh-narkar beats me to it.